plugins/autofarm.cpp

```diff
@@ -168,41 +162,37 @@ class AutoFarm {
     {
         plantable_plants.clear();
 
-        map<int, int> counts;
+        std::map<int, int> counts;
 
-        df::item_flags bad_flags;
-        bad_flags.whole = 0;
-
-#define F(x) bad_flags.bits.x = true;
-        F(dump); F(forbid); F(garbage_collect);
-        F(hostile); F(on_fire); F(rotten); F(trader);
-        F(in_building); F(construction); F(artifact);
+        const uint32_t bad_flags{
+#define F(x) (df::item_flags::Mask::mask_##x)
+        F(dump) | F(forbid) | F(garbage_collect) |
+        F(hostile) | F(on_fire) | F(rotten) | F(trader) |
+        F(in_building) | F(construction) | F(artifact)
 #undef F
+        };
 
-        for (auto ii : world->items.other[df::items_other_id::SEEDS])
+        for (auto& ii : world->items.other[df::items_other_id::SEEDS])
```

Comment:
Is there a benefit to using a reference to a pointer here?

Reply:
not really, but at the same time there's no real cost to it

Reply:
Fair enough